tests/upgrade/postgres_run.sh

```diff
@@ -234,8 +234,14 @@ test_upgrade_paths() {
     kubectl -n stackrox set image deploy/sensor "*=$REGISTRY/main:$CURRENT_TAG"
     kubectl -n stackrox set image deploy/admission-control "*=$REGISTRY/main:$CURRENT_TAG"
     kubectl -n stackrox set image ds/collector "collector=$REGISTRY/collector:$(make collector-tag)" \
-        "compliance=$REGISTRY/main:$CURRENT_TAG" \
-        "node-inventory=$REGISTRY/scanner-slim:$(make scanner-tag)"
+        "compliance=$REGISTRY/main:$CURRENT_TAG"
+    # Check if collector has 2 or 3 containers - we expect to see 3 only on Openshift 4
+    if kubectl -n stackrox get ds/collector -o=jsonpath='{$.spec.template.spec.containers[2].image}'; then
```

Comment:
It seems a bit fragile to use the number of containers alone as an indicator for doing this. A better way to check this would be to match container names. You can output all container names and then check as a substring.

Here's an example with JSONPath (untested)
```bash
if [[ $(kubectl -n stackrox get ds/collector -o=jsonpath='{$.spec.template.spec.containers[*].name}') == *"node-inventory"* ]]; then
    # update image
fi
```

Reply:
Addressed in [53410c9](https://github.com/stackrox/stackrox/pull/5422/commits/53410c9714483a0c7b37e418e04b21c97d25aa2f)